text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,927 @@
+        tip2MessageId: 78d546b46aec4557872139a48f66bc567687e8413578a14323548732358914a2
+          
+    'get-tips-response-example-same':
+      value:
+        tip1MessageId: f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d
+        tip2MessageId: f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d
+    
+    'post-messages-response-example':
+      value:
+        messageId: 7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006
+    
+    'get-messages-by-index-response-example':
+      value:
+        index: someindexkey
+        maxResults: 1000
+        count: 2
+        messageIds:
+          - f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d
+          - 7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006
+    
+    'get-message-by-id-response-example-new':
+      value:
+        messageId: f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d
+        parent1MessageId: 7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006
+        parent2MessageId: d97e89e49ed263b763afeda192515fe154ec826ceb92ef08926cd68ec02c9510
+        solid: false
+        referencedByMilestone: null
+        shouldPromote: true
+        shouldReattach: false
+    
+    'get-message-by-id-response-example-solid':
+      value:
+        messageId: f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d
+        parent1MessageId: 7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006
+        parent2MessageId: d97e89e49ed263b763afeda192515fe154ec826ceb92ef08926cd68ec02c9510
+        solid: true
+        referencedByMilestone: null
+        shouldPromote: false
+        shouldReattach: false
+    
+    'get-message-by-id-response-example-included':
+      value:
+        messageId: f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d
+        parent1MessageId: 7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006
+        parent2MessageId: d97e89e49ed263b763afeda192515fe154ec826ceb92ef08926cd68ec02c9510
+        solid: true
+        referencedByMilestone: 15465
+        ledgerInclusionState: included
+    
+    'get-message-by-id-response-example-conflicting':
+      value:
+        messageId: f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d
+        parent1MessageId: 7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006
+        parent2MessageId: d97e89e49ed263b763afeda192515fe154ec826ceb92ef08926cd68ec02c9510
+        solid: true
+        referencedByMilestone: 15465
+        ledgerInclusionState: conflicting
+
+    'get-message-by-id-data-response-example':
+      value:
+        hash: 82b1f7ba715d23874168f132eaa53cb7e9860cc84bca092b64e23b228bf1b8e2
+        version: 1
+        parent1: '0000000000000000000000000000000000000000000000000000000000000000'
+        parent2: '0000000000000000000000000000000000000000000000000000000000000000'
+        payload:
+          type: 0
+          transaction:
```

Comment:
Should say `essence`

Reply:
I added a PR with these changes and the requested `data` and `error` structures
https://github.com/rufsam/protocol-rfcs/pull/2

Reply:
See [https://editor.swagger.io/?url=https://raw.githubusercontent.com/alexsporn/protocol-rfcs/master/text/0026-rest-api/rest-api.yaml](Rendered Version)